samples/manage-features/src/main/java/com/esri/arcgismaps/sample/managefeatures/components/ManageFeaturesViewModel.kt

```diff
@@ -0,0 +1,338 @@
+            snackBarMessage = "Layer not loaded!"
+            return
+        }
+        when (currentFeatureOperation) {
+            manageFeaturesList[0] -> createFeatureAt(singleTapConfirmedEvent.screenCoordinate)
+            manageFeaturesList[1] -> deleteFeatureAt(singleTapConfirmedEvent.screenCoordinate)
+            manageFeaturesList[2] -> selectFeatureForAttributeEditAt(singleTapConfirmedEvent.screenCoordinate)
+            manageFeaturesList[3] -> updateFeatureGeometryAt(singleTapConfirmedEvent.screenCoordinate)
+        }
+    }
+
+    /**
+     * Set the current feature operation to perform based on the selected index from the dropdown. Also, reset feature
+     * selection.
+     */
+    fun onFeatureOperationSelected(index: Int) {
+        currentFeatureOperation = manageFeaturesList[index]
+        // Reset the selected feature when the operation changes.
+        damageLayer?.clearSelection()
+        selectedFeature = null
+    }
+
+    /**
+     * Create a new feature at the tapped location with some default attributes
+     */
+    private fun createFeatureAt(screenCoordinate: ScreenCoordinate) {
+        // Create the feature.
+        val feature = damageFeatureTable?.createFeature()?.apply {
+            // Get the normalized geometry for the tapped location and use it as the feature's geometry.
+            mapViewProxy.screenToLocationOrNull(screenCoordinate)?.let { mapPoint ->
+                geometry = GeometryEngine.normalizeCentralMeridian(mapPoint)
+                // Set feature attributes.
+                attributes["typdamage"] = "Minor"
+                attributes["primcause"] = "Earthquake"
+            }
+        }
+        feature?.let {
+            viewModelScope.launch {
+                // Add the feature to the table.
+                damageFeatureTable?.addFeature(it)
+                // Apply the edits to the service on the service geodatabase.
+                damageFeatureTable?.serviceGeodatabase?.applyEdits()
+                // Update the feature to get the updated objectid - a temporary ID is used before the feature is added.
+                it.refresh()
+                // Confirm feature addition.
+                snackBarMessage = "Created feature ${it.attributes["objectid"]}"
+            }
+        }
+    }
+
+    /**
+     * Selects a feature at the tapped location in preparation for deletion.
+     */
+    private fun deleteFeatureAt(screenCoordinate: ScreenCoordinate) {
+        damageLayer?.let { damageLayer ->
+            // Clear any existing selection.
+            damageLayer.clearSelection()
+            selectedFeature = null
+            viewModelScope.launch {
+                // Determine if a user tapped on a feature.
+                mapViewProxy.identify(damageLayer, screenCoordinate, 10.dp).onSuccess { identifyResult ->
+                    selectedFeature = (identifyResult.geoElements.firstOrNull() as ArcGISFeature).also {
```

Comment:
Sample crashes if I select a blank area of the map to identify and `deleteFeatureAt`: 
```suggestion
                    selectedFeature = (identifyResult.geoElements.firstOrNull() as? ArcGISFeature).also {
```